deploy/awstools/awstools.py

```diff
@@ -158,8 +183,19 @@ def get_aws_userid():
     But it seems that by default many accounts do not have permission to run this,
     so instead we get it from instance metadata.
     """
-    res = local("""curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | grep -oP '(?<="accountId" : ")[^"]*(?=")'""", capture=True)
-    return res.stdout.lower()
+    res = None
+    # This takes multiple minutes without a timeout from a non-AWS instance. In
+    # practice it should resolve nearly instantly on an initialized EC2 instance.
+    curl_connection_timeout = 10
+    with settings(warn_only=True), hide('everything'):
+        res = local("""curl -s --connect-timeout {} http://169.254.169.254/latest/dynamic/instance-identity/document | grep -oP '(?<="accountId" : ")[^"]*(?=")'""".format(curl_connection_timeout), capture=True)
+
+    # Use some default string if we're not on an EC2 instance (e.g., when a
+    # manager is launched from CI; during demos; other etc)
+    if res.return_code != 0:
+        return "PREAWSINIT"
```

Comment:
This should probably just error like in #1037 when the error code isn't 0, 28

Reply:
Yes.  I figured #1037 supercedes this and you'll take care of it.